Sources/Core/Filters/Inline/InlineFilterView.swift

```diff
@@ -97,11 +107,25 @@ private extension InlineFilterView {
             segments.append(segment)
         }
 
-        for titles in segmentTitles {
-            let segment = Segment(titles: titles)
-            segment.addTarget(self, action: #selector(handleValueChanged(segment:)), for: .valueChanged)
-            segments.append(segment)
+        if let filter = filter {
```

Comment:
Can we keep `InlineFilterView` unaware of selection store, filter, vertical and other model-related types? If the lines below are the ones that fix inline filter selection, they could perhaps go to `RootFilterViewController`.

Reply:
Sure, fixing it ;)

Reply:
Changed it back now. I also moved the `didChangeSelection` delegate call into `didSelectFilter` because the inline filter would remove and set multiple filter, which resulted in multiple search requests. Now it will only be one request. Had to do some changes in `FreeTextFilterViewController` to use the same method.